src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,329 @@
+    List<StaticLangApiMethodView> apiMethods = new ArrayList<>();
+
+    for (Method method : context.getNonStreamingMethods()) {
+      MethodConfig methodConfig = context.getMethodConfig(method);
+      MethodTransformerContext methodContext = context.asMethodContext(method);
+
+      if (methodConfig.isPageStreaming()) {
+        /*if (methodConfig.isFlattening()) {
+          for (ImmutableList<Field> fields : methodConfig.getFlattening().getFlatteningGroups()) {
+            apiMethods.add(
+                apiMethodTransformer.generatePagedFlattenedMethod(methodContext, fields));
+          }
+        }
+        apiMethods.add(apiMethodTransformer.generatePagedRequestObjectMethod(methodContext));
+        apiMethods.add(apiMethodTransformer.generatePagedCallableMethod(methodContext));
+        apiMethods.add(apiMethodTransformer.generateUnpagedListCallableMethod(methodContext));*/
+      } else {
+        if (methodConfig.isFlattening()) {
+          for (ImmutableList<Field> fields : methodConfig.getFlattening().getFlatteningGroups()) {
+            apiMethods.add(transformFlattenedMethod(apiMethodTransformer.generateFlattenedMethod(methodContext, fields, ApiMethodType.FlattenedMethod)));
+            apiMethods.add(transformFlattenedMethod(apiMethodTransformer.generateFlattenedMethod(methodContext, fields, ApiMethodType.FlattenedMethodAsyncCallSettings)));
+            apiMethods.add(transformFlattenedMethod(apiMethodTransformer.generateFlattenedMethod(methodContext, fields, ApiMethodType.FlattenedMethodAsyncCancellationToken)));
+          }
+        }
+        //apiMethods.add(apiMethodTransformer.generateRequestObjectMethod(methodContext));
+        //apiMethods.add(apiMethodTransformer.generateCallableMethod(methodContext));
+      }
+    }
+
+    return apiMethods;
+  }
+  
+  private StaticLangApiMethodView transformFlattenedMethod(StaticLangApiMethodView method) {
+    switch (method.type())
+    {
+      case FlattenedMethod:
+        method = AddParam(method, "CallSettings", "callSettings", "null", "If not null, applies overrides to this RPC call.");
+        method = AddReturnDoc(method, "The RPC response.");
+        break;
+      case FlattenedMethodAsyncCallSettings:
+        method = AddParam(method, "CallSettings", "callSettings", "null", "If not null, applies overrides to this RPC call.");
+        method = MakeAsync(method);
+        method = AddReturnDoc(method, "A Task containing the RPC response.");
+        break;
+      case FlattenedMethodAsyncCancellationToken:
+        method = AddParam(method, "CancellationToken", "cancellationToken", null, "A <see cref=\"CancellationToken\"/> to use for this RPC.");
+        method = MakeAsync(method);
+        method = AddReturnDoc(method, "A Task containing the RPC response.");
+        break;
+      default:
+        throw new RuntimeException("Unexpected method type: '" + method.type() + "'");
+    }
+    return method;
```

Comment:
I am uneasy with this pattern of unsealing/rebuilding view model objects. It makes it more difficult to trace the inputs and outputs of the transformation, because the set of values seen on a final view model object would be provided at multiple places in the code. This is almost equivalent to just having mutable view model classes. I would much prefer to add parameters into generateFlattenedMethod so that it could do all of the necessary work.